core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/localscan/LocalCsvFileScanOpDesc.java

```diff
@@ -86,17 +82,102 @@ public Schema sourceSchema() {
     }
 
     private Schema inferSchema(String headerLine) {
+        java.util.List<AttributeType> attributeType = IntStream
+                .range(0, headerLine.split(delimiter).length).mapToObj(i->AttributeType.INTEGER).collect(Collectors.toList());
+
+
+        try(BufferedReader reader = new BufferedReader(new FileReader(filePath))) {
+            String line;
+            if (header != null && header)
+                reader.readLine();
+            int i = 0;
+            while((line = reader.readLine())!=null && i<=100) {
+                String[] records = line.split((delimiter));
+                attributeType = inferline(attributeType, records);
+                i++;
+            }
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        java.util.List<AttributeType> finalAttributeType = attributeType;
+        System.out.println("a->"+finalAttributeType);
+//        System.out.println(Schema.newBuilder().add(IntStream.range(0, headerLine.split(delimiter).length).
+//                mapToObj(i -> new Attribute(l.get(i), finalAttributeType.get(i)))
+//                .collect(Collectors.toList())).build());
+
         if (delimiter == null) {
             return null;
         }
         if (header != null && header) {
-            return Schema.newBuilder().add(Arrays.stream(headerLine.split(delimiter)).map(c -> c.trim())
-                    .map(c -> new Attribute(c, AttributeType.STRING)).collect(Collectors.toList())).build();
+
+
+            return Schema.newBuilder().add(IntStream.range(0, headerLine.split(delimiter).length).
+                    mapToObj(i -> new Attribute(headerLine.split(delimiter)[i], finalAttributeType.get(i)))
+                    .collect(Collectors.toList())).build();
+//            return Schema.newBuilder().add(Arrays.stream(headerLine.split(delimiter)).map(c -> c.trim())
+//                    .map(c -> new Attribute(c, AttributeType.STRING)).collect(Collectors.toList())).build();
         } else {
             return Schema.newBuilder().add(IntStream.range(0, headerLine.split(delimiter).length).
-                    mapToObj(i -> new Attribute("column" + i, AttributeType.STRING))
+                    mapToObj(i -> new Attribute("column" + i, finalAttributeType.get(i)))
                     .collect(Collectors.toList())).build();
         }
     }
+    private java.util.List<AttributeType> inferline (java.util.List<AttributeType> attributeType, String[] line) {
+        return IntStream.range(0, line.length).
+                mapToObj(i->inferChar(attributeType.get(i),line[i])).collect(Collectors.toList());
+    }
+
+    private AttributeType inferChar (AttributeType attributeType, String c) {
```

Comment:
1. The logic of this block of code is unclear to me. Why do you need to take in attributeType (s) to do the inference?
2. The name of this method is misleading. the input String c is not a char, but a word/token?

Reply:
1. Sorry for misleading. This attributeType is the current inferred attributeType for the whole column. This is base on infer from previous row. It can only upgrade in Integer->Long->Double->Boolean->String/TimeStamp. This is necessary to keep tracking the attributeType of column.
2. I change it to `private AttributeType inferToken (AttributeType attributeType, String token) {`